src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java

```diff
@@ -290,9 +412,167 @@ public WindowTownHall(final BuildingTownHall.View townHall)
         registerButton(BUTTON_CHANGE_SPEC, this::doNothing);
         registerButton(BUTTON_TOGGLE_JOB, this::toggleHiring);
 
+        registerButton(BUTTON_PREV_PAGE_PERM, this::switchPage);
+        registerButton(BUTTON_NEXT_PAGE_PERM, this::switchPage);
+
+        registerButton(BUTTON_MANAGE_OFFICER, this::editOfficer);
+        registerButton(BUTTON_MANAGE_FRIEND, this::editFriend);
+        registerButton(BUTTON_MANAGE_NEUTRAL, this::editNeutral);
+        registerButton(BUTTON_MANAGE_HOSTILE, this::editHostile);
+
+
         registerButton(BUTTON_UP, this::updatePriority);
         registerButton(BUTTON_DOWN, this::updatePriority);
         registerButton(BUTTON_DELETE, this::deleteWorkOrder);
+
+        findPaneOfTypeByID(BUTTON_PREV_PAGE_PERM, Button.class).setEnabled(false);
+        findPaneOfTypeByID(BUTTON_MANAGE_OFFICER, Button.class).setEnabled(false);
+
+        registerButton(BUTTON_TRIGGER, this::trigger);
+
+        registerButton(BUTTON_ADD_BLOCK, this::addBlock);
+    }
+
+    /**
+     * Called when the "addBlock" button has been triggered.
+     * Tries to add the content of the input field as block or position to the colony.
+     */
+    private void addBlock()
+    {
+        //todo really have the lists in the colony view and add the new blocks here then!
+        final TextField input = findPaneOfTypeByID(INPUT_BLOCK_NAME, TextField.class);
+        final String inputText = input.getText();
+
+        final Block block = Block.getBlockFromName(inputText);
+
+        if(block != null)
+        {
+            MineColonies.getNetwork().sendToServer(new AddFreeToInteractBlockMessage(townHall.getColony(), block.getDefaultState()));
+        }
+
+        final String[] strings = inputText.split(" ");
+
+        if(strings.length == 3)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Assign this magic number 3 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)